### ISSUE TITLE
Netlify Deployment Error - This sounds like I know what I'm talking about...

### ISSUE THREAD

**Issue by lcnichols** [lcnichols/100daysofcode#2]
"be sure to add it in a code block so it stands out" - hum, not sure what that means? I'll click on this icon above that says "insert code" that seems logical at this point :D I also don't know how much of the log I should paste so hopefully this is right.

Holy Fork! Can't do @ mentions or assigns yet until you accept it seems.

`7:46:31 PM: [build:eleventy] Problem writing Eleventy templates: (more in DEBUG output)
7:46:31 PM: [build:eleventy] > TemplateLayoutPathResolver directory does not exist for page: _includes
7:46:31 PM: [build:eleventy]
7:46:31 PM: [build:eleventy] `Error` was thrown:
7:46:31 PM: [build:eleventy]     Error: TemplateLayoutPathResolver directory does not exist for page: _includes
7:46:31 PM: [build:eleventy]         at TemplateLayoutPathResolver.findFileName (/opt/build/repo/node_modules/@11ty/eleventy/src/TemplateLayoutPathResolver.js:101:13)
7:46:31 PM: [build:eleventy]         at TemplateLayoutPathResolver.init (/opt/build/repo/node_modules/@11ty/eleventy/src/TemplateLayoutPathResolver.js:68:28)
7:46:31 PM: [build:eleventy]         at new TemplateLayoutPathResolver (/opt/build/repo/node_modules/@11ty/eleventy/src/TemplateLayoutPathResolver.js:21:10)
7:46:31 PM: [build:eleventy]         at new TemplateLayout (/opt/build/repo/node_modules/@11ty/eleventy/src/TemplateLayout.js:16:24)
7:46:31 PM: [build:eleventy]         at Function.getTemplate (/opt/build/repo/node_modules/@11ty/eleventy/src/TemplateLayout.js:44:16)
7:46:31 PM: [build:eleventy]         at Template.getLayout (/opt/build/repo/node_modules/@11ty/eleventy/src/Template.js:86:37)
7:46:31 PM: [build:eleventy]         at Template.getData (/opt/build/repo/node_modules/@11ty/eleventy/src/Template.js:242:27)
7:46:31 PM: [build:eleventy]         at async Template.getTemplateMapEntries (/opt/build/repo/node_modules/@11ty/eleventy/src/Template.js:744:16)
7:46:31 PM: [build:eleventy]         at async TemplateMap.add (/opt/build/repo/node_modules/@11ty/eleventy/src/TemplateMap.js:32:21)
7:46:31 PM: [build:eleventy]         at async Promise.all (index 3)
7:46:31 PM: [build:eleventy] Wrote 0 files in 0.11 seconds (v0.11.0)
7:46:31 PM: [build:eleventy] npm ERR! code ELIFECYCLE
7:46:31 PM: [build:eleventy] npm ERR! errno 1
7:46:31 PM: [build:eleventy] npm ERR! sr-11ty-tailwind-alpine-template@1.0.0 build:eleventy: `ELEVENTY_ENV=production eleventy`
7:46:31 PM: [build:eleventy] npm ERR! Exit status 1
7:46:31 PM: [build:eleventy] npm ERR!
7:46:31 PM: [build:eleventy] npm ERR! Failed at the sr-11ty-tailwind-alpine-template@1.0.0 build:eleventy script.
7:46:31 PM: [build:eleventy] npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
7:46:31 PM: [build:eleventy]
7:46:31 PM: [build:eleventy] npm ERR! A complete log of this run can be found in:
7:46:31 PM: [build:eleventy] npm ERR!     /opt/buildhome/.npm/_logs/2020-12-03T00_46_31_579Z-debug.log
7:46:31 PM: ERROR: "build:eleventy" exited with 1.
7:46:31 PM: npm ERR! code ELIFECYCLE
7:46:31 PM: npm ERR! errno 1
7:46:31 PM: npm ERR! sr-11ty-tailwind-alpine-template@1.0.0 build: `run-s clean build:* --print-label`
7:46:31 PM: npm ERR! Exit status 1
7:46:31 PM: npm ERR!
7:46:31 PM: npm ERR! Failed at the sr-11ty-tailwind-alpine-template@1.0.0 build script.
7:46:31 PM: npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
7:46:31 PM: npm ERR! A complete log of this run can be found in:
7:46:31 PM: npm ERR!     /opt/buildhome/.npm/_logs/2020-12-03T00_46_31_594Z-debug.log`

**Comment by shanerobinson** [comment]
@lcnichols : Without forking and testing myself, my guess is this is related to Issue #1 and at the very least you should add an `.eleventy.js` file that specifies the **input** and **output** dirs as described in the first code block here: https://www.11ty.dev/docs/config/

Here is the relevant part of the error log above: 

`7:46:31 PM: [build:eleventy] > TemplateLayoutPathResolver directory does not exist for page: _includes`

**Comment by lcnichols** [comment]
Ah ok I just read this after I did my `.eleventy.js` file. This is cool, I'm running deployment now (which I'm guessing is the same meaning as "publish"?

**Comment by lcnichols** [comment]
OMG deployment was successful! However, I get a Page Not Found when I click on my [URL](https://lcn-100daysofcode.netlify.app)

**Comment by shanerobinson** [comment]
Yep... It's still related to issue #1 and your `input` directory value. 😉 

## Deploy -vs- Publish

You **deploy** a site. 
You **publish** a blog post or article or new entry.

**Comment by shanerobinson** [comment]
PS: You're running `npm run dev` locally while you're working on the site, yes? You're not waiting to commit and deploy to see if the site is working...?

**Comment by lcnichols** [comment]
Um, I haven't typed `npm run dev` yet today...so I'm guessing not. What does that do again?

**Comment by lcnichols** [comment]
This is what happens when I did the `npm run dev` command

`> sr-11ty-tailwind-alpine-template@1.0.0 dev
> npm-run-all clean --parallel dev:* --print-label

sh: npm-run-all: command not found
npm ERR! code 127
npm ERR! path /Users/lananichols/Sites/100daysofcode
npm ERR! command failed
npm ERR! command sh -c npm-run-all clean --parallel dev:* --print-label

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/lananichols/.npm/_logs/2020-12-03T18_07_58_292Z-debug.log`

**Comment by lcnichols** [comment]
Okay, reading about NPM now...so much to learn.

**Comment by shanerobinson** [comment]
Yep... My guess is you didn't run `npm install` after you cloned the repo to your local machine. So none of the dependencies in the `package.json` file were installed and so the entire project can't run locally. 

You develop locally by running `npm run dev` _(or whatever the scripts in `package.json` are)_. 

Generally the README.md file has instructions, provided by the owner, on how to install and run local dev. 

I'll continue to help like this if you prefer to slog through using a cloned or forked repo -vs- building a simple 11ty project from scratch. I think building from scratch is easier and you'll learn more, faster. Your call...

**Comment by lcnichols** [comment]
Let's start from scratch. I just bought 2 books, I need to do some lingo and basic learning so I can actually understand things, haha, go figure!

**Comment by shanerobinson** [comment]
Awesome! It'll make more sense on the other side... I use the metaphor of learning to play an instrument... Everyone just wants to sit down at a piano or pick up a guitar and learn to play a song. But it doesn't work that way if you don't know where Middle C is, where to put your fingers, or even basic lingo.  😄

**Comment by lcnichols** [comment]
Chris ran the `npm install` and all that jazz last night trying to help me, it was kinda sexy even though I still didn't know what he was doing haha. I don't know how you're getting emoji's in here and a rocket ship...I will find out soon :D

**Comment by shanerobinson** [comment]
1. Emojis = start typing a colon ":" followed by what emoji you want until you see it, then arrow down to select it, then Return to place it.  This feature also works in some other Mac apps that support emojis. Other wise, on a Mac you can hit "Control + Command + Spacebar" and if you have the Emoji keyboard active on your Mac, it'll popup the Emoji selector modal. If you **don't** have the Emoji keyboard active, you can go into **System Prefs > Keyboard** and check the "Show keyboard and emoji viewes in menu bar" check box. 

2. Note the **"M⬇️ "** at the lower right below the comment window below? Hover it and it pops a "Tool Tip" that says Markdown is supported" so you can use Markdown to style these comments instead of having to click the formatting buttons above the comment box. 

3. If you thought it was sexy watching Chris type in the terminal, you understand why it makes Roxanne "moist" that I can be typing something, while looking at her, and talking to her about something different all at the same time. 😀

**Comment by lcnichols** [comment]
# I practiced a little markdown last night

🤠 Ah, I have the Control Strip on my Mac so it took me a few minutes, I typed colon then have to hit the emoji on my control touch strip on my keyboard lol

🥰 this one was w/o the colon, okay it's always there I just never look at my keyboard haha!

OMG you and Chris with the multitasking abilities!!! BAAAAHAHAHAHAHA!
